### ISSUE TITLE
Trying to use latexindent in VSCode

### ISSUE THREAD

**Issue by Nbelles** [cmhughes/latexindent.pl#339]
Please provide the following when posting an issue:

I'm not sure exactly what the source of what might be causing this issue but I believe I have it setup in VSCode such that it runs latexindent on file save. It attempts to run latexindent but fails according to the LaTeX Workshop output (as seen in the output below). Walking through the pieces of the command, I do have latexindent installed at that location `/usr/local/texlive/2021basic/texmf-dist/scripts/latexindent/latexindent.pl`. 

I tried opening a terminal on my computer and opening that directory and running the latexindent.pl using `./latexindent.pl` it give the error: `zsh: exec format error: latexindent` even though running `perl latexindent.pl` works just fine `Please enter text to be indented: (press CTRL+D when finished)`.

I have a feeling it is potentially wrong with my latexindent install because when I try running it from the terminal it gives me that error shown above, otherwise I would think that it might be something wrong with the LaTeX Workshop module of VSCode. 

Thanks for any insight. 

original .tex code
--
```
\documentclass{article}

\title{Test}
\author{Test 123}
\date{\today}

\begin{document}

\maketitle

\end{document}
```

yaml settings
--
N/A

actual/given output
--
From LaTeX Workshop output tab in VSCode:
[19:23:48] Start formatting with latexindent.
[19:23:48] Formatting with command /usr/local/texlive/2021basic/texmf-dist/scripts/latexindent/latexindent -c="/Users/Nathaniel/" "/Users/Nathaniel/__latexindent_temp.tex" -y=defaultIndent: '    '
[19:23:48] Formatting failed: spawn /usr/local/texlive/2021basic/texmf-dist/scripts/latexindent/latexindent EACCES
[19:23:48] stderr: 
[19:23:48] Formatting failed with exit code -13
[19:23:48] stderr: 

desired or expected output
--
N/A

anything else
--
N/A

**Comment by cmhughes** [comment]
Try the following commands from the terminal, stop when you get the first one that works:

latexindent
latexindent.pl
perl latexindent.pl

**Comment by cmhughes** [comment]
latexindent should be available from any directory, as it should be in your path. You shouldn't have to use the leading "./" as in

./latexindent.pl

**Comment by Nbelles** [comment]
Here is the output from running the following commands:
```
[computer:~]
[Nathaniel]% latexindent
zsh: exec format error: latexindent
[computer:~]
[Nathaniel]% latexindent.pl
zsh: command not found: latexindent.pl
[computer:~]
[Nathaniel]% perl latexindent.pl
Can't open perl script "latexindent.pl": No such file or directory
[computer:~]
[Nathaniel]% cd /usr/local/texlive/2021basic/texmf-dist/scripts/latexindent/ 
[computer:latexindent]
[Nathaniel]% perl latexindent.pl
Please enter text to be indented: (press CTRL+D when finished)
[computer:latexindent]
[Nathaniel]% 
```

**Comment by cmhughes** [comment]
This looks like it's a problem with your path.

Is your texlive path in your PATH variable? The first option should work fine.

**Comment by Nbelles** [comment]
How would I check that?

**Comment by cmhughes** [comment]
On Linux I think the command is

echo $PATH

I think it's similar on Mac,
https://stackoverflow.com/questions/14617041/how-can-i-see-the-current-value-of-my-path-variable-on-os-x#:~:text=You%20need%20to%20use%20the,contents%20as%20a%20command%20name
.

**Comment by Nbelles** [comment]
Here's what my path looks like right now: `/opt/local/bin:/opt/local/sbin:/Library/Frameworks/Python.framework/Versions/3.8/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/opt/X11/bin:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Applications/Wireshark.app/Contents/MacOS:/Applications/Visual Studio Code.app/Contents/Resources/app/bin:/Users/Nathaniel/Library/Python/3.8/bin`

**Comment by cmhughes** [comment]
I don't see the texlive path you posted previously in that path.

**Comment by Nbelles** [comment]
Which path do I need to add to it? Just `/usr/local/texlive` or `/usr/local/texlive/2021basic/texmf-dist/scripts` or `/usr/local/texlive/2021basic/texmf-dist/scripts/latexindent`?

**Comment by cmhughes** [comment]
Does your pdflatex work from the terminal...? What steps do you normally
follow to get your texlive working?

**Comment by Nbelles** [comment]
Yes, running `pdflatex test.tex` (the same test.tex file as before) works just fine.

**Comment by cmhughes** [comment]
Can you add usr/local/texlive/2021basic/texmf-dist/scripts to your path?

**Comment by cmhughes** [comment]
I'm closing this, let me know if it isn't resolved.

**Comment by Nbelles** [comment]
Sorry I wasn't able to get back to this until just now. 

I've added that directory to my path and I get the error `zsh: permission denied: latexindent`

**Comment by Nbelles** [comment]
I just tried doing a `which latexindent` to see if my terminal knows where it exists and it returns `latexindent not found`

**Comment by Nbelles** [comment]
@cmhughes Just tagging you in case you didn't see the notifications because this is closed.

**Comment by cmhughes** [comment]
I'm afraid I don't have much to add to what I've said already in this
thread already.

The problem appears to be one centered on your path variable. You need to
get your path variable set up so that includes the directory that
latexindent is installed in.

This varies from operating system to operating system. For example, on
Linux I can edit my bashrc file.

It's not clear to me if every user of your operating system has the same
issue. I would hope not and hope that I would have heard from others prior
to now.

In summary : you need to find a way to update your path variable to include
the directory latexindent is installed in.

**Comment by cmhughes** [comment]
Which user does latexindent.pl belong to? Maybe it has been installed as root and needs to have its permissions changed?
